contracts/libraries/Math.sol

```diff
@@ -80,6 +97,288 @@ library Math {
         }
     }
 
+    /// @notice Returns the index of the most significant bit of the number,
+    ///     where the least significant bit is at index 0 and the most significant bit is at index 255
+    /// @dev The function satisfies the property:
+    ///     x >= 2**mostSignificantBit(x) and x < 2**(mostSignificantBit(x)+1)
+    /// @param x the value for which to compute the most significant bit, must be greater than 0
+    /// @return r the index of the most significant bit
+    function mostSignificantBit(uint256 x) internal pure returns (uint8 r) {
+        require(x > 0);
+
+        if (x >= 0x100000000000000000000000000000000) {
+            x >>= 128;
+            r += 128;
+        }
+        if (x >= 0x10000000000000000) {
+            x >>= 64;
+            r += 64;
+        }
+        if (x >= 0x100000000) {
+            x >>= 32;
+            r += 32;
+        }
+        if (x >= 0x10000) {
+            x >>= 16;
+            r += 16;
+        }
+        if (x >= 0x100) {
+            x >>= 8;
+            r += 8;
+        }
+        if (x >= 0x10) {
+            x >>= 4;
+            r += 4;
+        }
+        if (x >= 0x4) {
+            x >>= 2;
+            r += 2;
+        }
+        if (x >= 0x2) r += 1;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                TICK MATHS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Calculates 1.0001^(tick/2) as an X96 number
+    /// @dev Implemented using Uniswap's "incorrect" constants. Supplying commented-out real values for an accurate calculation
+    /// @dev Will revert if |tick| > max tick
+    /// @param tick Value of the tick for which sqrt(1.0001^tick) is calculated
+    /// @return sqrtPriceX96 A Q64.96 number representing the sqrt price at the provided tick
+    function getSqrtRatioAtTick(int24 tick) internal pure returns (uint160 sqrtPriceX96) {
```

Comment:
I verified this to be 100% compatible with Uniswap's implementation across the tick range. It returns the same result on every tick and is consistently more gas efficient, so great job on that! There might be further optimizations, but this is good enough for now.